server/cluster/metadata/topology_manager.go

```diff
@@ -261,6 +270,14 @@ func (m *TopologyManagerImpl) RemoveTable(ctx context.Context, shardID storage.S
 		}
 	}
 
+	for i, oldShardView := range m.shardViews {
+		if oldShardView.ShardID == shardID {
+			m.shardViews[i] = m.shardViews[len(m.shardViews)-1]
+			m.shardViews = append(m.shardViews, newShardView)
+			break
+		}
+	}
```

Comment:
Avoid clone the shard view when looping it:
```suggestion
	for i := range m.shardViews {
		if m.shardViews[i].ShardID == shardID {
			m.shardViews[i] = newShardView
			break
		}
	}
```

And how to deal with case where the shard is not found in the shardViews?

BTW, maybe we can define it as a method for reuse here and L206.